python/cudf/cudf/pylibcudf_tests/test_lists.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import pyarrow as pa
+import pytest
+from utils import assert_column_eq
+
+from cudf._lib import pylibcudf as plc
+
+
+def test_concatenate_rows():
+    test_data = [[[0, 1], [2], [5], [6, 7]], [[8], [9], [], [13, 14, 15]]]
+
+    arrow_tbl = pa.Table.from_arrays(test_data, names=["a", "b"])
+    plc_tbl = plc.interop.from_arrow(arrow_tbl)
```

Comment:
In general we aim to use module-scope fixtures for these kinds of test inputs to maximize reuse and minimize the number of independent memory allocations (which speeds up test runs). It's not critical in this case since you only use this once, but something to keep in mind for future pylibcudf PRs.